wagtail/admin/tests/test_userbar.py

```diff
@@ -63,6 +63,55 @@ def test_userbar_tag_anonymous_user(self):
         # Make sure nothing was rendered
         self.assertEqual(content, '')
 
+    def test_add_item_without_template_context(self):
+        from wagtail.core import hooks
+
+        class UserbarKittenLinkItemFromContext:
+            def render(self, request):
+                return '<li><a href="www.kittenwar.com" ' \
+                       + 'target="_parent" class="action icon icon-wagtail">Kittens!</a></li>'
+
+        @hooks.register('construct_wagtail_userbar')
+        def add_kitten_link_item(request, items):
+            return items.append(UserbarKittenLinkItemFromContext())
+
+        template = Template("{% load wagtailuserbar %}{% wagtailuserbar %}")
+        ctx = Context({
+            PAGE_TEMPLATE_VAR: self.homepage,
+            'request': self.dummy_request(self.user),
+        })
+        content = template.render(ctx)
+        self.assertIn("www.kittenwar.com", content)
+
+    def test_add_item_with_template_context(self):
```

Comment:
I'm gonna try and figure something out :)

Reply:
I don't know how to create simple test, because I must to add object with "render" function, and i have to remove hook, but i didn't found simple way to delete hook-function. Easiest way is to delete test.

Reply:
The kind of thing I have in mind is more of a 'unit test' approach. We already know from other tests that:
- The `{% wagtailuserbar %}` tag is rendered when included in a template
- The `{% wagtailuserbar %}` tag passes the items to `construct_wagtail_userbar` to allow them to be modified

All we really need to show in these new tests is:
- When an function registered with `construct_wagtail_userbar` doesn't accept a `context` argument, when `wagtailuserbar` is called, a deprecation warning is produced
- When a function registered with `construct_wagtail_userbar` does accept a `context` argument, when `wagtailuserbar` is called, the context is passed through to it.

We could do this simply by importing `wagtailuserbar` from `templatetags` and calling it like any normal Python function. The function registered with `construct_wagtail_userbar` that accepts `context` doesn't even need to do anything with the values in the context to prove anything - we just need to know that is receives it. This could be as simple as setting a `context_received` variable to True if receives the `context`, e.g. 

```
def test_registered_function_receive_context_if_accepted(self):
    mock_context = {'foo': 'bar'}
    context_received = False
    
   @hooks.register('construct_wagtail_userbar')
   def modify_userbar(request, items, context=None):
       context_received = context is not None
       
   try:
       wagtailuserbar(mock_context)
   finally:
       # always deregister the function
       hook_list = hooks._hooks['construct_wagtail_userbar']
       hooks._hooks['construct_wagtail_userbar'] = [i for i in hook_list if i[0].__name__ != 'modify_userbar']
    
    # Confirm that our function received the context
    self.assertTrue(context_received)
```
What do you think?